clang/lib/Sema/SemaExpr.cpp

```diff
@@ -10555,6 +10555,43 @@ static void checkArithmeticNull(Sema &S, ExprResult &LHS, ExprResult &RHS,
       << LHS.get()->getSourceRange() << RHS.get()->getSourceRange();
 }
 
+static void DetectPrecisionLossInComplexDivision(Sema &S, QualType DivisorTy,
+                                                 SourceLocation OpLoc) {
+  // Either real/real or complex/real division.
+  // Either way there can be no precision loss.
+  auto *CT = DivisorTy->getAs<ComplexType>();
+  if (!CT)
+    return;
```

Comment:
Hmm, is this right? If this is the *divisor*, not the *dividend*, then isn’t this just complex/complex division? Because I don’t think real/complex is valid (at least I’d expect us to convert the dividend to a complex number first).

Reply:
I noticed we were passing the *LHS* expression to this, but now we’re passing in the *RHS* type? Is that intended?

Reply:
Changing to RHS (divisor) is intended. If the divisor is real (**not** `isAnyComplexType`) its either complex / real or real / real divison. I think the code is right. I also double checked the names of the terms of division because English isn't my primary language, but that seems right too.

Reply:
That makes sense; looking at this again, I think the way the comment is worded just confused me a bit.